buildSrc/src/main/kotlin/org/gradle/modules/ClasspathManifestPatcher.kt

```diff
@@ -0,0 +1,96 @@
+package org.gradle.modules
+
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.artifacts.ResolvedDependency
+import org.gradle.util.CollectionUtils
+import org.gradle.util.GUtil
+
+import org.gradle.kotlin.dsl.*
+
+import org.objectweb.asm.ClassReader
+import org.objectweb.asm.ClassWriter
+import org.objectweb.asm.commons.ClassRemapper
+import org.objectweb.asm.commons.Remapper
+
+import java.io.File
+
+
+// Map of internal APIs that have been renamed since the last release
+val REMAPPINGS = mapOf("org/gradle/plugin/use/internal/PluginRequests" to "org/gradle/plugin/management/internal/PluginRequests")
+
+open class ClasspathManifestPatcher(val project: Project, val temporaryDir: File, val runtime: Configuration, val moduleNames: Set<String>) {
```

Comment:
:x: The signature is too long. A line break before the first parameter would be good.